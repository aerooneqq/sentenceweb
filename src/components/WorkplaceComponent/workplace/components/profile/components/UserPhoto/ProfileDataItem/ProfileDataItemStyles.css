.profileDataItem{
  width: 38px;
  height: 38px;

  background: orangered;

  border-width: 1px;
  border-color: orangered;
  border-style: solid;
  border-radius: 40px;

  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
}

.profileDataItem:hover{
  background: orange;
  border-color: orange;
}

.profileDataItem:nth-child(1){
  transform: translate(0, -20px);
}
.profileDataItem:nth-child(2){
  transform: translate(-3px, 5px);
}
.profileDataItem:nth-child(3){
  transform: translate(0px, 15px);
}
.profileDataItem:nth-child(4){
  transform: translate(3px, 5px);
}
.profileDataItem:nth-child(5){
  transform: translate(0px, -20px);
}

.profileDataItemImg{
  width: 25px;
  height: 25px;

  background: transparent;

  border-style: none;
}
