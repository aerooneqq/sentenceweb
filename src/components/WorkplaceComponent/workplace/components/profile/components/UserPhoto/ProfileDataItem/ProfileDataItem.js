import React, {Component} from "react"

//Styles
import "./ProfileDataItemStyles.css"

export default class ProfileDataItem extends Component{
  constructor(props){
    super(props)

    this.handleClick = this.handleClick.bind(this)
  }

  handleClick() { 
    this.props.data.changeUserData(this.props.data.currentUserData)
  }

  render(){
    return(
      <div className = "profileDataItem" onClick={this.handleClick}>
        <img src = {this.props.icon} className = "profileDataItemImg" alt = "" />
      </div>
    )
  }
}
