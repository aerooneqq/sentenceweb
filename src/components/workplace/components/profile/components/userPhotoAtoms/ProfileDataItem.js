import React, {Component} from "react"
import "./styles/ProfileDataItem.css"

export default class ProfileDataItem extends Component{
  constructor(props){
    super(props)

    this.state = { 
      imgURL: this.props.imgURL
    }
  }

  render(){
    return(
      <div className = "profileDataItem">
        <img src = {this.state.imgURL} className = "profileDataItemImg">
        </img>
      </div>
    )
  }
}
