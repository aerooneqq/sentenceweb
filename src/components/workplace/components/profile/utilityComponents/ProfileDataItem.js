import React, {Component} from "react"

import "../styles/ProfileDataItem.css"

export default class ProfileDataItem extends Component{
  constructor(){
    super()
  }

  render(){
    return(
      <div className = "profileDataItem">
        <div className = "profileDataItemImg">

        </div>
      </div>
    )
  }
}
