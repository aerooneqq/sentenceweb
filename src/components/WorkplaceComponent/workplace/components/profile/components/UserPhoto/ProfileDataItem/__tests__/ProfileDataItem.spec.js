import React from 'react'
import { shallow, render } from 'enzyme'
import {shallowToJson} from "enzyme-to-json"
import ProfileDataItem from "../ProfileDataItem";

describe("Profile data item", () => { 
    it("should render correctly", () => { 
        const profileDataItem = shallow(<ProfileDataItem />); 

        expect(shallowToJson(profileDataItem)).toMatchSnapshot();
    });
});