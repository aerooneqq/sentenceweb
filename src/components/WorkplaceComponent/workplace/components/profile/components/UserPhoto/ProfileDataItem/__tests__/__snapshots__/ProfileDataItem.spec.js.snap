// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile data item should render correctly 1`] = `
<div
  className="profileDataItem"
  onClick={[Function]}
>
  <img
    alt=""
    className="profileDataItemImg"
  />
</div>
`;
